libsolidity/experimental/analysis/FunctionCallGraph.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsolidity/experimental/analysis/Analysis.h>
+#include <libsolidity/experimental/analysis/FunctionCallGraph.h>
+
+using namespace solidity::frontend::experimental;
+using namespace solidity::util;
+
+FunctionCallGraph::FunctionCallGraph(solidity::frontend::experimental::Analysis& _analysis):
+	m_analysis(_analysis),
+	m_errorReporter(_analysis.errorReporter()),
+	m_currentNode(nullptr),
+	m_inFunctionDefinition(false)
```

Comment:
Isn't this rather
```suggestion
	m_currentFunction(nullptr)
```

And ``m_inFunctionDefinition == (m_currentFunction != nullptr)``, always?

Reply:
I mean, yes, but only if I set `m_currentFunction` to `nullptr` in `endVisit`, right? In any case, I'd still need to keep toggling `m_inFunctionDefinition`, and `m_inFunctionDefinition = true/false` I find preferable.

Reply:
Although, setting `m_currentFunction` to `nullptr` in `endVisit` should actually be done anyway.

Reply:
> Although, setting `m_currentFunction` to `nullptr` in `endVisit` should actually be done anyway.

Yes - and then another boolean flag is redundant.

Reply:
Yes, it's redundant, but also easier to read, which is why I'm in favour of keeping it. I can get rid of it if you want to though.

Reply:
Get rid of it.

Reply:
If you see it gone, you'll also stop disagreeing with getting rid of it, I'm sure :-).
You never want to unnecessarily keep two copies of the same state around that you'll need to manage separately.